src/components/ProgressBar/ProgressBar.jsx

```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import { Flex, Progress, Text, useMantineTheme } from "@mantine/core";
+
+export const ProgressBar = () => {
+  const { colors } = useMantineTheme();
+
+  const [timerValue, setTimerValue] = React.useState(0);
+
+  let timerId = setTimeout(() => {
+    setTimerValue(timerValue + 1);
+  }, 1000);
+
+  if (timerValue >= 30) {
+    clearTimeout(timerId);
+  }
+
+  const getProgressBarValue = () => Math.ceil(((30 - timerValue) * 100) / 30);
+
+  const getTime = () => {
+    const timeValue = 30 - timerValue;
```

Comment:
(Optional, not necessary)

You're using `30 - timerValue` twice, so it could be a separate variable too!

Reply:
These are the different functions. I'll leave my variant